bonito/openvino/model.py

```diff
@@ -0,0 +1,93 @@
+import os
+import numpy as np
+import torch
+
+try:
+    from openvino.inference_engine import IECore, StatusCode
+    from .loader import torch2openvino
+except ImportError:
+    pass
+
+class OpenVINOModel:
+
+    def __init__(self, model, half, dirname):
+        self.model = model
+        self.alphabet = model.alphabet
+
+        model_name = model.config['model'] + ('_fp16' if half else '')
+        xml_path, bin_path = [os.path.join(dirname, model_name) + ext for ext in ['.xml', '.bin']]
+        self.ie = IECore()
+        if os.path.exists(xml_path) and os.path.exists(bin_path):
+            self.net = self.ie.read_network(xml_path, bin_path)
+        else:
+            self.net = torch2openvino(model)
+        self.exec_net = None
+
+
+    def eval(self):
+        pass
+
+
+    def half(self):
+        return self
+
+
+    def to(self, device):
+        self.device = str(device).upper()
+
+
+    def __call__(self, data):
+        data = np.expand_dims(data, axis=2)  # 1D->2D
+        batch_size = data.shape[0]
+        if not self.exec_net:
+            inp_shape = list(data.shape)
+            inp_shape[0] = 1  # We will run the batch asynchronously
+            self.net.reshape({'input': inp_shape})
```

Comment:
Does it suppose to change rank of the network?

Reply:
We determine network input shape only at first call